### DISCUSSION TITLE
Problem with the Input System

### DISCUSSION THREAD

**Discussion by Onemario1234** [Leather128/LeathersFunkinEngine#25] [General]
The input system, while otherwise good, is not good for playing with additional keys.

I'm a novice who plays the Shaggy mod quite a bit, and I have some experience with 9 keys. Because the Leather Engine's input system punishes you for hitting a note too early, it becomes really tough to hit notes in 9 key songs. And when I do hit notes, I feel like I get punished for doing so, and it just gets too frustrating too quickly. The Psych Engine also has a penalty system when you hit a note too early, but it isn't as brutal.

**Comment by Leather128** [comment]
i get what your talking about personally, but the only reason it is so punishing is to prevent from spamming. i will lower the amount of damage to a little bit below a miss, but personally i just dont want people spamming, and antimash systems never seemed to feel fair in the end for me personally

**Reply by Onemario1234** [reply]
I understand where you're coming from, but when it comes to Friday Night Funkin', I feel people should be able to play however they want.

And besides that, I don't want to end up spamming during 9 key songs, but punishing my current playstyle this much is not encouraging me to try playing better. It's discouraging, more than anything. My current playstyle is trash, it works, but it's trash. And when that old playstyle suddenly stops working, I lose motivation to relearn.

**Comment by Onemario1234** [comment]
Well, how about this as a compromise?: Make it an option. Those who want to play with the antimash system can turn it on. Those who don't, can turn it off.

Because, after thinking about it more, I DO see it's a good learning tool to have. But I can still see it as frustrating from time to time, and I'm sure others might feel the same.

**Reply by Leather128** [reply]
i think this would be a good comprimise, i'll add this soon after i get done with dealing with some other things lol (issues and a pull request)

**Reply by Leather128** [reply]
this has now been implemented under the setting name "Anti Mash" in the gameplay options.

**Reply by Onemario1234** [reply]
Sweet! Glad we could come to that compromise!
